app/src/main/java/com/github/sdpteam15/polyevents/model/database/local/dao/GenericEntityDao.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.sdpteam15.polyevents.model.database.local.dao
+
+import androidx.room.*
+import com.github.sdpteam15.polyevents.model.database.local.entity.EventLocal
+import com.github.sdpteam15.polyevents.model.database.local.entity.GenericEntity
+
+/**
+ * Data access object for the GenericEntity entity on the local room database
+ */
+@Dao
+interface GenericEntityDao {
+
+    @Query("SELECT * FROM entity_table WHERE collection = :collection")
+    suspend fun getAll(collection: String): List<GenericEntity>
+
+    @Query("SELECT * FROM entity_table WHERE id = :id AND collection = :collection ")
+    suspend fun get(id: String, collection: String): GenericEntity
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insert(event: GenericEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertAll(events: List<GenericEntity>)
+
+    @Delete
+    suspend fun delete(event: EventLocal)
+
+    @Query("DELETE FROM entity_table")
+    suspend fun deleteAll()
+
+    @Query("SELECT MAX(update_time) FROM entity_table WHERE collection = :collection")
+    suspend fun lastUpdate(collection: String): String?
+}
```

Comment:
You should add some comments for the description of each method

Reply:
You don't have to add comments for all them, maybe for `lastUpdate`. The rest is self explanatory I think.

Reply:
Maybe also explain what the Generic Entity is and what will you use it for

Reply:
Yes, you right. So just explain the goal and the role of the class !

Reply:
It's done